### PR TITLE
More changes

### LINKED ISSUE #166: Add the possibility to reload a specific plugin (or all plugins)

### LINKED ISSUE #167: pluginapi reload/list commands are not registered
probably because they are using their command API for some strange reason since the game has its own API.

image

### LINKED ISSUE #169: Cant find checkpoint doors.
Trying to lock these doors at the start of the round.
lcz_checkpoint
```
[PluginEvent(ServerEventType.RoundStart)]
void OnRoundStart()
{
    List<FacilityDoor> checkpoint_doors = Facility.Doors.FindAll((door) => door.Permissions.HasFlag(KeycardPermissions.Checkpoints));
    foreach (FacilityDoor door in checkpoint_doors)
    {
        door.Lock(DoorLockReason.AdminCommand, true);
    }

    ServerConsole.AddLog(checkpoint_doors.Count.ToString() + " doors have the checkpoint permission out of " + Facility.Doors.Count.ToString());
}
```
```
0 doors have the checkpoint permission out of 92
```

The problem is the same for LightZone.Doors

### PR DESCRIPTION
**Pull request by SrLicht**
I'm sorry but I'm here to break this number.
image

# Round.cs

* Added ``IsRoundEnded``. Please add ``RoundSummary.singleton.RoundEnded => RoundSummary.singleton._roundEnded``

# FacilityDoor
* Added Name variable

Resolves #169 

# Commands

* Fixed the NWAPI commands that were not registered.

* Now you can reload the plugins configuration.

* Now you can reload the gameplay configuration.

fix #167 and resolves #166 

# Player.cs

* Fixed a bug that ArtificialHealth gave when trying to obtain or change the AHP of a player who is SCP.

* Fixed a bug when trying to use ArtificialHealth since it was being incorrectly invoked.

# Map.cs

*  Added  ``Map.Cameras``, ``Map.PocketDimensionTeleports``, ``Map.Lockers``, ``Map.Elevators``, ``Map.TeslaGates``, ``Map.Generators``

* Added ``Map.Broadcast(ushort duration, string message, Broadcast.Flag flag, bool clearPrevius = false)``

* Added ``Map.ClearBroadcast()``

* Added ``Map.ForceDecontamination()``

* Added ``enum Map.DecontaminationStatus``

* Added ``Map.GetRandomRoom(FacilityZone zone)`` return RoomIdentifier or null

* Added ``Map.FlickerLights(float duration, FacilityZone zone)``

* Added ``Map.FlickerLights(float duration, IEnumerable<FacilityZone> zones)``

* Added ``Map.FlickerAllLights(float duration)``

* Added ``Map.TurnOnAllLights()``

* Added ``Map.TurnOnLights(FacilityZone zone)``

* Added ``Map.TurnOnLights(IEnumerable<FacilityZone> zones)``

* Added ``Map.ChangeColorOfAllLights(Color color, float intensity = 0)``

* Added ``Map.ChangeColorOfLights(Color color, FacilityZone zone, float intensity = 0)``

* Added ``Map.ChangeColorOfLights(Color color, IEnumerable<FacilityZone> zones, float intensity = 0)``

* Added ``Map.ResetColorOfAllLights``

* Added ``Map.ResetColorOfLights(FacilityZone zone)``

### CODE REVIEW

**Comment by SrLicht** [comment]
```cs
internal static List<Scp079Camera> _cameras = new(150); //Idk the exact number of cameras per map.

internal static List<PocketDimensionTeleport> _teleports = new(8);

internal static List<Locker> _lockers = new(100);
```
These are concepts of what I am trying to do, since at first I was thinking of starting them from the NWAPI itself but I would have to create a class just for that, so I preferred to consult with a draft here

**Comment by moddedmcplayer** [comment]
Pretty sure there are already some lists for some of these, otherwise you can always just replace it with => Object.FindObjectsOfType<T>();

**Review by Killers0992 — APPROVED** [review]

File: NwPluginAPI/Commands/ReloadCommand.cs [hunk 1]
```diff
@@ -6,25 +11,50 @@ namespace PluginAPI.Commands
...
+			if (arguments.IsEmpty())
 			{
-				plugins.Add($"<color=lime>{(plugin.PluginName)}</color>");
+				response = $"Please specify a valid argument\nUsage: plugins reload {this.DisplayCommandUsage()}";
+				return false;
 			}
 
-			response = $"{(plugins.Count == 0 ? "Reloaded 0 plugin configs!" : $"Reloaded {string.Join(", ", plugins)} plugin configs!")}";
-			return true;
+			switch (arguments.At(0).ToLower())
```
Comment (zabszk) [inline NwPluginAPI/Commands/ReloadCommand.cs:32]:
```suggestion
			switch (arguments.At(0).ToLowerInvariant())
```

File: NwPluginAPI/Core/Doors/FacilityGate.cs [hunk 2]
```diff
@@ -22,6 +22,11 @@ public class FacilityGate : FacilityDoor
 		/// </summary>
 		public new static int Count => List.Count;
 
+		/// <summary>
+		/// Gets gate name.
+		/// </summary>
+		public string Name => OriginalObject.name;
```
Comment (zabszk) [inline NwPluginAPI/Core/Doors/FacilityGate.cs:28]: Is this needed? `FacilityGate` inherits from `FacilityDoor` and you already added `Name` there.

**Review by zabszk — CHANGES_REQUESTED** [review]

File: NwPluginAPI/Core/Doors/FacilityGate.cs (hunk 2 above)
Comment (SrLicht) [reply NwPluginAPI/Core/Doors/FacilityGate.cs:28]: You are right

Comment (SrLicht) [reply NwPluginAPI/Core/Doors/FacilityGate.cs]: Fixed in commit [da48294](https://github.com/northwood-studios/NwPluginAPI/pull/161/commits/da482942d286f980f8287b966f4c54babb84248a)

**Review by zabszk — APPROVED** [review]

**Pull request merged**